<a href="https://colab.research.google.com/github/Amazile/imobiliaria/blob/main/Respondendo_questionamentos_do_diretor_da_imobili%C3%A1ria.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1. Importação das bibliotecas

In [ ]:
import pandas            as pd

#2. Carregamento dos dados

In [ ]:
df=pd.read_csv('/content/aluguel.csv',sep=';')

#3. Descrição dos dados

In [ ]:
df

Tipo                Bairro  ...  Condominio    IPTU
0                    Quitinete            Copacabana  ...       500.0    60.0
1                         Casa       Jardim Botânico  ...         NaN     NaN
2      Conjunto Comercial/Sala       Barra da Tijuca  ...      4020.0  1111.0
3                  Apartamento                Centro  ...       390.0    20.0
4                  Apartamento          Higienópolis  ...       230.0     NaN
...                        ...                   ...  ...         ...     ...
32955                Quitinete                Centro  ...       350.0    25.0
32956              Apartamento           Jacarepaguá  ...       800.0    40.0
32957              Apartamento  São Francisco Xavier  ...       509.0    37.0
32958              Apartamento                Leblon  ...       760.0     NaN
32959  Conjunto Comercial/Sala                Centro  ...      4206.0  1109.0

[32960 rows x 9 columns]

In [ ]:
#Informações sobre as variáveis dos imóveis 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32960 entries, 0 to 32959
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Tipo        32960 non-null  object 
 1   Bairro      32960 non-null  object 
 2   Quartos     32960 non-null  int64  
 3   Vagas       32960 non-null  int64  
 4   Suites      32960 non-null  int64  
 5   Area        32960 non-null  int64  
 6   Valor       32943 non-null  float64
 7   Condominio  28867 non-null  float64
 8   IPTU        22723 non-null  float64
dtypes: float64(3), int64(4), object(2)
memory usage: 2.3+ MB


In [ ]:
df.shape
print('A base de dados apresenta {} registros/imóveis e {} informações/variáveis.'.format(df.shape[0],df.shape[1]))

A base de dados apresenta 32960 registros/imóveis e 9 informações/variáveis.


In [ ]:
df.head()

Tipo           Bairro  ...  Condominio    IPTU
0                Quitinete       Copacabana  ...       500.0    60.0
1                     Casa  Jardim Botânico  ...         NaN     NaN
2  Conjunto Comercial/Sala  Barra da Tijuca  ...      4020.0  1111.0
3              Apartamento           Centro  ...       390.0    20.0
4              Apartamento     Higienópolis  ...       230.0     NaN

[5 rows x 9 columns]

In [ ]:
df_tipo=df[['Tipo']]
df_tipo

Tipo
0                    Quitinete
1                         Casa
2      Conjunto Comercial/Sala
3                  Apartamento
4                  Apartamento
...                        ...
32955                Quitinete
32956              Apartamento
32957              Apartamento
32958              Apartamento
32959  Conjunto Comercial/Sala

[32960 rows x 1 columns]

#4. Respostas ao diretor da imobiliária

In [ ]:
#tipos de imóveis encontrados na carteira imobiliária
df_tipo.drop_duplicates()

Tipo
0                        Quitinete
1                             Casa
2          Conjunto Comercial/Sala
3                      Apartamento
7               Casa de Condomínio
16                  Prédio Inteiro
17                            Flat
29                      Loja/Salão
80         Galpão/Depósito/Armazém
83                  Casa Comercial
117                   Casa de Vila
159                 Terreno Padrão
207                    Box/Garagem
347                           Loft
589    Loja Shopping/ Ct Comercial
2157                       Chácara
3354         Loteamento/Condomínio
4379                         Sítio
4721                 Pousada/Chalé
6983                        Studio
9687                         Hotel
23614                    Indústria

In [ ]:
#Quais são os 5 principais tipos de imóveis encontrados na carteira da imobiliária e respectivas quantidades?
df_tipo.value_counts()

Tipo                       
Apartamento                    19532
Conjunto Comercial/Sala         6815
Loja/Salão                      1426
Casa de Condomínio               996
Casa                             967
Quitinete                        836
Galpão/Depósito/Armazém          623
Flat                             476
Prédio Inteiro                   301
Casa Comercial                   265
Casa de Vila                     249
Loja Shopping/ Ct Comercial      247
Box/Garagem                       82
Terreno Padrão                    70
Loft                              51
Sítio                             10
Loteamento/Condomínio              5
Studio                             4
Hotel                              2
Chácara                            1
Indústria                          1
Pousada/Chalé                      1
dtype: int64

###Os 5 tipos de imóveis mais encontados na imobiliária são: apartamento (19.532), conjunto comercial/sala (6.815), loja/salão (1.426), casa de condomínio (996) e casa (967). 

In [ ]:
df_tipo.value_counts().sort_index()

Tipo                       
Apartamento                    19532
Box/Garagem                       82
Casa                             967
Casa Comercial                   265
Casa de Condomínio               996
Casa de Vila                     249
Chácara                            1
Conjunto Comercial/Sala         6815
Flat                             476
Galpão/Depósito/Armazém          623
Hotel                              2
Indústria                          1
Loft                              51
Loja Shopping/ Ct Comercial      247
Loja/Salão                      1426
Loteamento/Condomínio              5
Pousada/Chalé                      1
Prédio Inteiro                   301
Quitinete                        836
Studio                             4
Sítio                             10
Terreno Padrão                    70
dtype: int64

In [ ]:
#bairros encontrados
df_bairro=df[['Bairro']]
df_bairro

Bairro
0                Copacabana
1           Jardim Botânico
2           Barra da Tijuca
3                    Centro
4              Higienópolis
...                     ...
32955                Centro
32956           Jacarepaguá
32957  São Francisco Xavier
32958                Leblon
32959                Centro

[32960 rows x 1 columns]

In [ ]:
#Quais são os cinco bairros com mais imóveis para locação?
df_bairro.value_counts()

Bairro                  
Barra da Tijuca             6259
Centro                      3258
Copacabana                  3186
Ipanema                     2233
Recreio dos Bandeirantes    2127
                            ... 
Cidade Jardim                  1
Cidade de Deus                 1
Parque Colúmbia                1
Grumari                        1
Itaguaí                        1
Length: 162, dtype: int64

###Os bairros com mais imóveis para locação são: Barra da Tijuca, Centro, Copacabana, Ipanema e Recreio dos Bandeirantes. 

In [ ]:
#Quantos imóveis apresentam no mínimo 4 quartos e aluguel menor que R$2.000,00?
df_especial=(df['Quartos']>= 4) & (df['Valor']<2000.0)
df_especial_quartos_valor=df[df_especial].shape[0]
df_especial_quartos_valor

43

In [ ]:
#Criar coluna com o valor bruto do aluguel (valor+condomínio+IPTU)
df['Valor bruto aluguel']=df['Valor']+df['Condominio']+df['IPTU']
df.head(10)

Tipo           Bairro  ...    IPTU  Valor bruto aluguel
0                Quitinete       Copacabana  ...    60.0               2260.0
1                     Casa  Jardim Botânico  ...     NaN                  NaN
2  Conjunto Comercial/Sala  Barra da Tijuca  ...  1111.0              10331.0
3              Apartamento           Centro  ...    20.0               1210.0
4              Apartamento     Higienópolis  ...     NaN                  NaN
5              Apartamento     Vista Alegre  ...     NaN                  NaN
6              Apartamento         Cachambi  ...    17.0               1618.0
7       Casa de Condomínio  Barra da Tijuca  ...     NaN                  NaN
8       Casa de Condomínio            Ramos  ...     NaN                  NaN
9  Conjunto Comercial/Sala           Centro  ...  3030.0              57223.0

[10 rows x 10 columns]

In [ ]:
#Criar coluna com valor/m2
df['Valor m2']=df['Valor']/df['Area']
df.head(10)

Tipo           Bairro  ...  Valor bruto aluguel   Valor m2
0                Quitinete       Copacabana  ...               2260.0  42.500000
1                     Casa  Jardim Botânico  ...                  NaN  70.000000
2  Conjunto Comercial/Sala  Barra da Tijuca  ...              10331.0  34.666667
3              Apartamento           Centro  ...               1210.0  53.333333
4              Apartamento     Higienópolis  ...                  NaN  16.666667
5              Apartamento     Vista Alegre  ...                  NaN  17.142857
6              Apartamento         Cachambi  ...               1618.0  26.000000
7       Casa de Condomínio  Barra da Tijuca  ...                  NaN  29.333333
8       Casa de Condomínio            Ramos  ...                  NaN  15.384615
9  Conjunto Comercial/Sala           Centro  ...              57223.0  50.359712

[10 rows x 11 columns]

In [ ]:
#arredondando valor m2 para duas casas decimais
df['Valor m2']=df['Valor m2'].round(2)
df.head()

Tipo           Bairro  ...  Valor bruto aluguel  Valor m2
0                Quitinete       Copacabana  ...               2260.0     42.50
1                     Casa  Jardim Botânico  ...                  NaN     70.00
2  Conjunto Comercial/Sala  Barra da Tijuca  ...              10331.0     34.67
3              Apartamento           Centro  ...               1210.0     53.33
4              Apartamento     Higienópolis  ...                  NaN     16.67

[5 rows x 11 columns]

In [ ]:
#selecionando casa, casa de condomínio e casa de vila
residencial_casa=['Casa','Casa de Condomínio', 'Casa de Vila']
selecao_casa=df['Tipo'].isin(residencial_casa)
dados_residencial_casa=df[selecao_casa]

In [ ]:
dados_residencial_casa

Tipo  ... Valor m2
1                    Casa  ...    70.00
7      Casa de Condomínio  ...    29.33
8      Casa de Condomínio  ...    15.38
18     Casa de Condomínio  ...    17.39
22     Casa de Condomínio  ...    16.09
...                   ...  ...      ...
32908  Casa de Condomínio  ...    12.90
32911                Casa  ...    23.33
32924  Casa de Condomínio  ...    20.00
32931                Casa  ...    31.25
32952  Casa de Condomínio  ...    33.33

[2212 rows x 11 columns]

In [ ]:
dados_residencial_casa.shape
print('A base de dados apresenta {} registros de casa/casa de condomínio/casa de vila.'.format(dados_residencial_casa.shape[0]))

A base de dados apresenta 2212 registros de casa/casa de condomínio/casa de vila.


In [ ]:
#selecionando apartamentos
residencial_apto=['Apartamento']
selecao_apto=df['Tipo'].isin(residencial_apto)
dados_residencial_apto=df[selecao_apto]

In [ ]:
dados_residencial_apto

Tipo                Bairro  ...  Valor bruto aluguel  Valor m2
3      Apartamento                Centro  ...               1210.0     53.33
4      Apartamento          Higienópolis  ...                  NaN     16.67
5      Apartamento          Vista Alegre  ...                  NaN     17.14
6      Apartamento              Cachambi  ...               1618.0     26.00
10     Apartamento                Centro  ...                  NaN     33.33
...            ...                   ...  ...                  ...       ...
32951  Apartamento        Vila Valqueire  ...                  NaN     19.23
32953  Apartamento                 Méier  ...               1438.0     12.86
32956  Apartamento           Jacarepaguá  ...               2640.0     23.08
32957  Apartamento  São Francisco Xavier  ...               1946.0     29.17
32958  Apartamento                Leblon  ...                  NaN     42.86

[19532 rows x 11 columns]

In [ ]:
dados_residencial_apto.shape
print('A base de dados apresenta {} registros de apartamentos.'.format(dados_residencial_apto.shape[0]))

A base de dados apresenta 19532 registros de apartamentos.
